### ISSUE TITLE
RuntimeError: CUDA error: invalid configuration argument

### ISSUE THREAD

**Issue by 644517236** [pytorch/pytorch#22526]
## Issue description

when it use  torch._C._nn.upsample_nearest2d(input, _output_size(2)), it's comes with 

RuntimeError: CUDA error: invalid configuration argument. 

The problem is only occur in torch-nightly( 1.2.0.dev20190702).

## Error log

File "/home/xdjf/XXXX/XXXXX/XXXXXXXX.py", line 74, in _upsample_add
    return F.interpolate(x, size=(H,W), mode='nearest') + y
File "/home/XXXX/.conda/envs/py36torch120/lib/python3.6/site-packages/torch/nn/functional.py", line 2545, in interpolate
    return torch._C._nn.upsample_nearest2d(input, _output_size(2))
RuntimeError: CUDA error: invalid configuration argument

**You can get the script and run it with:**
torch-nightly( 1.2.0.dev20190702)
```
import torch
import torch.nn.functional as F

x=torch.rand(4,256,1,1).cuda()
y=torch.rand(4,256,3,3).cuda()

F.interpolate(x, size=(3,3), mode='nearest') + y
```

- PyTorch:
- conda install pytorch-nightly cudatoolkit=10.0 -c pytorch
- OS:Ubuntu16.04 LTS
- PyTorch version: torch-nightly( 1.2.0.dev20190702)
- Python version:3.6
- CUDA/cuDNN version:9.0 or 10.0
- GPU models and configuration: GTX 1080ti * 4

**Comment by mrshenli** [comment]
cc @jjsjann123

**Comment by jjsjann123** [comment]
Probably related to my last update in nearest kernels. Let me take a look tomorrow.
Thanks for pinging me on this!

**Comment by jjsjann123** [comment]
This error is caused by launch config with block.z = 128. CUDA maximum z-dimension is 64 instead of 1024 (as with x/y).

I issued #22602 to fix this one.

**Comment by 644517236** [comment]
Thanks. @jjsjann123  @mrshenli

**Comment by zhanwenchen** [comment]
Hi @jjsjann123. This may be broken again. I just built from source (1.4.0) with CUDA 10.0 on NVIDIA 1060. Happened in Upsample scale_factor = (1, 2).


```python

DNNs/fcn_20191027202105757111_evaluating/scan_batteries/target_phantom_anechoic_cyst_2p5mm/target_3: r3: Denoising original stft with neural network model...
FlexNet: initial x.size() =  torch.Size([103040, 130])
FlexNet: after first view, x.size() =  torch.Size([103040, 1, 2, 65])
After prev_layer = Conv2d(1, 31, kernel_size=(2, 6), stride=(2, 2), padding=(0, 1)), x.size() = torch.Size([103040, 31, 1, 31]), x.mean() = -0.04480661079287529
After prev_layer = LeakyReLU(negative_slope=0.01), x.size() = torch.Size([103040, 31, 1, 31]), x.mean() = 0.11979887634515762
After prev_layer = Conv2d(31, 77, kernel_size=(2, 2), stride=(2, 2), padding=(1, 3)), x.size() = torch.Size([103040, 77, 1, 18]), x.mean() = -0.6838343143463135
After prev_layer = LeakyReLU(negative_slope=0.01), x.size() = torch.Size([103040, 77, 1, 18]), x.mean() = 0.052065860480070114
Traceback (most recent call last):
  File "evaluate_models.py", line 90, in <module>
    evaluate_one_model(model_folder)
  File "evaluate_models.py", line 47, in evaluate_one_model
    process_single_scan_battery(new_folder_name, scan_battery_dirname)
  File "/home/xxx/beam_nn/lib/process_single_scan_battery.py", line 56, in process_single_scan_battery
    process_single_target(target_dirname)
  File "/home/xxx/beam_nn/lib/process_single_scan_battery.py", line 104, in process_single_target
    new_stft_object = r3_dnn_apply(target_dirname, old_stft_obj=old_stft_obj, saving_to_disk=False)
  File "/home/xxx/beam_nn/lib/r3_dnn_apply.py", line 66, in r3_dnn_apply
    process_each_frequency(model_dirname, stft, frequency, using_cuda=using_cuda)
  File "/home/xxx/beam_nn/lib/r3_dnn_apply.py", line 155, in process_each_frequency
    aperture_data_new = model(aperture_data).cpu().data.numpy()
  File "/home/xxx/anaconda3/envs/pytorch-build/lib/python3.7/site-packages/torch/nn/modules/module.py", line 541, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/xxx/beam_nn/lib/flexnet.py", line 94, in forward
    x = self.net(x)
  File "/home/xxx/anaconda3/envs/pytorch-build/lib/python3.7/site-packages/torch/nn/modules/module.py", line 541, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/xxx/anaconda3/envs/pytorch-build/lib/python3.7/site-packages/torch/nn/modules/container.py", line 92, in forward
    input = module(input)
  File "/home/xxx/anaconda3/envs/pytorch-build/lib/python3.7/site-packages/torch/nn/modules/module.py", line 541, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/xxx/anaconda3/envs/pytorch-build/lib/python3.7/site-packages/torch/nn/modules/upsampling.py", line 131, in forward
    return F.interpolate(input, self.size, self.scale_factor, self.mode, self.align_corners)
  File "/home/xxx/anaconda3/envs/pytorch-build/lib/python3.7/site-packages/torch/nn/functional.py", line 2503, in interpolate
    return torch._C._nn.upsample_nearest2d(input, _output_size(2))
RuntimeError: CUDA error: invalid configuration argument

```

**Comment by wcf1065948474** [comment]
**This bug still exists in Pytorch 1.3.0**

```python
import torch
x=torch.rand(2048,256,8,8).cuda()
up=torch.nn.Upsample(scale_factor=2)
up(x).size()
```
Traceback (most recent call last):
  File "test.py", line 4, in <module>
    print(up(x).size())
  File "/usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py", line 541, in __call__
    result = self.forward(*input, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/torch/nn/modules/upsampling.py", line 131, in forward
    return F.interpolate(input, self.size, self.scale_factor, self.mode, self.align_corners)
  File "/usr/local/lib/python3.6/dist-packages/torch/nn/functional.py", line 2500, in interpolate
    return torch._C._nn.upsample_nearest2d(input, _output_size(2))
RuntimeError: CUDA error: invalid configuration argument


**But if you reduce the amount of data, it works fine**
```python
import torch
x=torch.rand(2048,256,4,4).cuda()
up=torch.nn.Upsample(scale_factor=2)
up(x).size() #output torch.Size([2048, 256, 8, 8])
```

**Comment by wcf1065948474** [comment]
```python
import torch
x=torch.rand(2048,256,8,8).cuda()
up=torch.nn.Upsample(scale_factor=2)
print(up(x).size())
```
Traceback (most recent call last):
  File "test.py", line 4, in <module>
    print(up(x).size())
  File "/usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py", line 541, in __call__
    result = self.forward(*input, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/torch/nn/modules/upsampling.py", line 131, in forward
    return F.interpolate(input, self.size, self.scale_factor, self.mode, self.align_corners)
  File "/usr/local/lib/python3.6/dist-packages/torch/nn/functional.py", line 2500, in interpolate
    return torch._C._nn.upsample_nearest2d(input, _output_size(2))
RuntimeError: CUDA error: invalid configuration argument

**But if you reduce the amount of data, it works fine**
```python
import torch
x=torch.rand(2048,256,4,4).cuda()
up=torch.nn.Upsample(scale_factor=2)
print(up(x).size())  #output torch.Size([2048, 256, 8, 8])
```

**Comment by jjsjann123** [comment]
Thanks for reporting and the repro.
Let me take a look at it.

**Comment by jjsjann123** [comment]
Same issue with the launch config, I should have done a better job with my last PR :man_facepalming:

Although I added striding on channel dimension, I forgot to limit the launch config on GPU properties (https://docs.nvidia.com/cuda/cuda-runtime-api/structcudaDeviceProp.html#structcudaDeviceProp_192d195493a9d36b2d827aaf3ffd89f1e)

Will tag my PR shortly.

**Comment by zhanwenchen** [comment]
@jjsjann123 you da best

**Comment by jjsjann123** [comment]
Quick update, the previous PR was botched... Round 2 PR: #29016

**Comment by gwgundersen** [comment]
It looks like #29016 was released in version 1.3.1? I am using that version

```
>>> import torch
>>> torch.__version__
'1.3.1'
```

however I see what looks to be the same error

```
...
  File "/scratch/gpfs/gwg3/dpcca_v8/models/celebaae.py", lcline 88, in decode
    t1 = F.interpolate(h1, scale_factor=2)
  File "/home/gwg3/.conda/envs/dpccav8/lib/python3.7/site-packages/torch/nn/functional.py", line 2500, in interpolate
    return torch._C.;(input, _output_size(2))
RuntimeError: CUDA error: invalid configuration argument
```

Is this a related bug or something else? Happy to provide more info if needed. FWIW, this code worked using a previous version of PyTorch.

**Comment by jjsjann123** [comment]
@gwgundersen Thanks a lot for reporting!

can you try it on master? I don't know if v1.3.1 tag is the actual one where 1.3.1 is built on, at least the fix is not there. https://github.com/pytorch/pytorch/commits/v1.3.1

Our last fix doesn't solve all the problem with the kernel, as the kernel is written in a way that doesn't stride on some dimensions. But we thought it's not a real use case. If you have a legit case there that breaks the kernel, a repro would be useful. (i.e. the size of `h1` in your script.)

**Comment by gwgundersen** [comment]
I tried the nightly build,

```
conda install pytorch torchvision cudatoolkit=10.1 -c pytorch-nightly
```

but now I run out of CUDA memory in what I assume is an unrelated bug—although oddly enough, I tried upgrading versions to 1.3.1 because I was running out of memory on 1.0.1. Sorry for my confusing reports. All this worked months ago, and I'm just context switching back into a dormant project.

Anyway, [here is the model](https://gist.github.com/gwgundersen/1c00a5e4dcc649c7e647d29bf25accc8#file-celebaae-py-L88). My input tensor has shape `(128, 3, 128, 128)`.
